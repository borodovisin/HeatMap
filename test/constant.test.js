
const util = require('../src/utils');

describe('Test for limitNumberSize constant', () => {
    test('Should be a number', () => {
        const type = typeof util.limitNumberSize;
        expect(type).toBe('number');
    });

    test('Should be greater than cero', () => {
        expect(util.limitNumberSize).toBeGreaterThan(0);
    });
});

describe('Test for getAxisOption function', () => {

    test('getAxisOption always return object', () => {
        const spy = jest.spyOn(util, 'getAxisOption');
        util.getAxisOption();
        expect(spy).toHaveReturned();
        expect(spy).toHaveReturnedWith(expect.any(Object));
    });

    it('Should return correct result', () => {
        const result = util.getAxisOption();
        expect(result).toMatchSnapshot();
    });
});
