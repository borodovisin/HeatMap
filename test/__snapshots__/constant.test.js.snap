// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for getAxisOption function Should return correct result 1`] = `
Object {
  "axisLabel": Object {
    "interval": 0,
  },
  "axisLine": Object {
    "show": false,
  },
  "axisTick": Object {
    "show": false,
  },
  "show": true,
  "splitArea": Object {
    "areaStyle": Object {
      "color": "transparent",
    },
    "show": true,
  },
  "splitLine": Object {
    "lineStyle": Object {
      "color": "#FFF",
      "width": 1,
    },
    "show": true,
  },
  "type": "category",
}
`;
